doc/dev-guide.md

```diff
@@ -458,6 +458,28 @@ sqldb.query(sql.select_questions, params, ...);
 SELECT * FROM questions WHERE id IN (SELECT unnest($id_list::INTEGER[]));
 ```
 
+* To pass a lot of data to SQL a useful pattern is to send a JSON object array and unpack it in SQL to the equivalent of a table. This is the pattern used by the "sync" code, such as [sprocs/sync_news_items.sql](https://github.com/PrairieLearn/PrairieLearn/blob/master/sprocs/sync_news_items.sql). For example, given the JavaScript:
+```javascript
+let data = [
+    {a: 5, b: "foo"},
+    {a: 9, b: "bar"}
+];
+let params = {data};
+sqldb.query(sql.insert_data, params, ...);
+```
+We can insert this into a DB table in a single operation with:
+```sql
+-- BLOCK insert_data
+INSERT INTO my_table (a, b, order_by)
+SELECT *
+FROM
+    ROWS FROM(
```

Comment:
Is the `ROWS FROM()` needed here since we’re only calling one function?

Reply:
It's needed for the `WITH ORDINALITY`. I split that out into a separate bullet point for clarity.